src/components/form/index.tsx

```diff
@@ -26,17 +28,28 @@ export class SmoothlyForm implements Changeable, Clearable, Submittable {
 	@Event() notice: EventEmitter<Notice>
 
 	@Prop({ mutable: true, reflect: true }) changed = false
-	private listeners: { changed?: ((parent: Changeable) => Promise<void>)[] } = {}
+	private listeners: {
+		changed?: ((parent: Changeable) => Promise<void>)[]
+		readonly?: ((parent: Editable) => Promise<void>)[]
+	} = {}
 
 	listen(property: "changed", listener: (parent: Changeable) => Promise<void>): void {
 		;(this.listeners[property] ??= []).push(listener)
 		listener(this)
 	}
+	listenReadonly(property: "readonly", listener: (parent: Editable) => Promise<void>): void {
```

Comment:
listen() and listenReadonly() do the same thing but i couldnt for the life of me figure out how to have one function do both. I involved @Magnus93 and he also hit the type-wall on this..